doc/services/storage/zms/zms.rst

```diff
@@ -0,0 +1,432 @@
+	struct zms_ate {
+		uint8_t crc8;      /* crc8 check of the entry */
+		uint8_t cycle_cnt; /* cycle counter for non erasable devices */
+		uint32_t id;       /* data id */
+		uint16_t len;      /* data len within sector */
+		union {
+			uint8_t data[8]; /* used to store small size data */
+			struct {
+				uint32_t offset; /* data offset within sector */
+				union {
+					uint32_t data_crc; /* crc for data */
+					uint32_t metadata; /* Used to store metadata information
+							    * such as storage version.
+							    */
+				};
+			};
+		};
+	} __packed;
+
+.. note:: The data CRC is checked only when the whole data of the element is read.
+   The data CRC is not checked for a partial read, as it is computed for the complete set of data.
+
+.. note:: Enabling the data CRC feature on a previously existing ZMS content without
+   data CRC will make all existing data invalid.
+
+.. _free-space:
+
+Available space for user data (key-value pairs)
+***********************************************
+
+For both scenarios ZMS should have always an empty sector to be able to perform the garbage
+collection.
+So if we suppose that 4 sectors exist in a partition, ZMS will only use 3 sectors to store
+Key-value pairs and keep always one (rotating sector) empty to be able to launch GC.
+
+.. note:: The maximum single data length that could be written at once in a sector is 64K
+   (This could change in future versions of ZMS)
+
+Small data values
+=================
+
+For small data values (<= 8 bytes), the data is stored within the entry (ATE) itself and no data
+is written at the top of the sector.
+ZMS has an entry size of 16 bytes which means that the maximum available space in a partition to
+store data is computed in this scenario as :
+
+.. math::
+
+   \small\frac{(NUM\_SECTORS - 1) \times (SECTOR\_SIZE - (5 \times ATE\_SIZE))}{2}
+
+Where:
+
+``NUM_SECTOR:`` Total number of sectors
+
+``SECTOR_SIZE:`` Size of the sector
+
+``ATE_SIZE:`` 16 bytes
+
+``(5 * ATE_SIZE):`` Reserved ATEs for header and delete items
+
+For example for 4 sectors of 1024 bytes, free space for data is :math:`\frac{3 \times 944}{2} = 1416 \, \text{ bytes}`.
+
+Large data values
+=================
+
+Large data values ( > 8 bytes) are stored separately at the top of the sector.
+In this case it is hard to estimate the free available space as this depends on the size of
+the data. But we can take into account that for N bytes of data (N > 8 bytes) an additional
+16 bytes of ATE must be added at the bottom of the sector.
+
+Let's take an example:
+
+For a partition that has 4 sectors of 1024 bytes and for data size of 64 bytes.
+Only 3 sectors are available for writes with a capacity of 944 bytes each.
+Each Key-value pair needs an extra 16 bytes for ATE which makes it possible to store 11 pairs
+in each sectors (:math:`\frac{944}{80}`).
+Total data that could be stored in this partition for this case is :math:`11 \times 3 \times 64 = 2112 \text{ bytes}`
+
+.. _wear-leveling:
```

Comment:
FYI, you don't need to define those when you only want to reference sections of your page from within the page itself. See e.g. [here](https://github.com/zephyrproject-rtos/zephyr/pull/76222/files#diff-aa3e5a0a44d82ae9e97891d29efb7f57a206b0804a1be69030403c5877495094R44) for an example.

This creates a global reference.
Which can be fine if it's properly named, e.g. prefixed by `zms_`. But with such a generic name it's not great.